Source/Core/Core/Lua/LuaFunctions/LuaBitFunctions.cpp

```diff
@@ -0,0 +1,143 @@
+#include "LuaBitFunctions.h"
+#include "../LuaVersionResolver.h"
+#include "common/CommonTypes.h"
+
+namespace Lua
+{
+namespace LuaBit
+{
+class BitClass
+{
+public:
+  BitClass() {}
+};
+
+BitClass* bitInstance;
+
+BitClass* GetBitInstance()
+{
+  if (bitInstance == NULL)
+    bitInstance = new BitClass();
+  return bitInstance;
+}
+
+  void InitLuaBitFunctions(lua_State* luaState, const std::string& luaApiVersion)
+  {
+    BitClass** bitPtrPtr = (BitClass**)lua_newuserdata(luaState, sizeof(BitClass*));
+    *bitPtrPtr = GetBitInstance();
+    luaL_newmetatable(luaState, "LuaBitMetaTable");
+    lua_pushvalue(luaState, -1);
+    lua_setfield(luaState, -2, "__index");
+
+    std::array luaBitFunctionsWithVersionsAttached =  {
+      luaL_Reg_With_Version({"bitwise_and", "1.0", bitwise_and}),
+      luaL_Reg_With_Version({"bitwise_or", "1.0", bitwise_or}),
+      luaL_Reg_With_Version({"bitwise_not", "1.0", bitwise_not}),
+      luaL_Reg_With_Version({"bitwise_xor", "1.0", bitwise_xor}),
+      luaL_Reg_With_Version({"logical_and", "1.0", logical_and}),
+      luaL_Reg_With_Version({"logical_or", "1.0", logical_or}),
+      luaL_Reg_With_Version({"logical_xor", "1.0", logical_xor}),
+      luaL_Reg_With_Version({"logical_not", "1.0", logical_not}),
+      luaL_Reg_With_Version({"bit_shift_left", "1.0", bit_shift_left}),
+      luaL_Reg_With_Version({"bit_shift_right", "1.0", bit_shift_right})
+    };
+
+    std::unordered_map<std::string, std::string> deprecatedFunctionsMap = std::unordered_map<std::string, std::string>();
```

Comment:
This can be simplified as:

```suggestion
    std::unordered_map<std::string, std::string> deprecatedFunctionsMap;
```

the default constructor is always executed, so the assignment can be omitted.